subsys/bluetooth/host/hci_core.c

```diff
@@ -5189,7 +5238,7 @@ int bt_le_adv_start(const struct bt_le_adv_param *param,
 		return err;
 	}
 
-	if (!(param->options & BT_LE_ADV_OPT_ONE_TIME)) {
+	if (!(param->options & BT_LE_ADV_OPT_ONE_TIME) && !dir_adv) {
```

Comment:
if BT_LE_ADV_OPT_ONE_TIME is set implicitly when dir_adv is assigned to true, we dont need the additional !dir_adv check.

Reply:
Problem with this is that we allow `param` to be const, i.e. we'd need to have a local copy of it (probably not worth  it)

Reply:
If we really want to take this implicit approach, it is also possible to make a copy of ```param``` in the ```bt_conn_create_slave_le``` function and set proper options

Reply:
So what is the consensus over this ?
1) leave it as it is
2) copy ```param``` to the local variable in the ```bt_le_adv_start_internal``` and set the implicit options if it is directed advertising mode
3) copy ```param``` to the local variable in the ```bt_conn_create_slave_le ``` and set the implicit options

Reply:
I'd go for 1, since it's quite easy to add the extra condition to `bt_le_adv_start_internal()` and `valid_adv_param()`, same way as was done for ONE_TIME.

Reply:
`param->options` is referenced 8 times (indirect addressing) in this function, I will go for an auto variable `options = param->options;`, `if (dir_adv) { options |= (...);} /* set one time and connectable flags */`, and use `options` in the other 8 places.

Reply:
I'm ok for bt_le_adv_start_internal(). As for valid_adv_param() I think it'd be overkill there - you don't have the same 8 references etc that @cvinayak was talking about.

Reply:
That said, another option could be to create the auto variable higher up the stack, i.e. immediately in the call from the application, and then pass it together with the param to start_internal and valid_adv_param.

Reply:
Mainly just brainstorming here without more carefully looking at the resulting code - do what feels good and I'll take a look at the next iteration of your patch :)

Reply:
> That said, another option could be to create the auto variable higher up the stack, i.e. immediately in the call from the application, and then pass it together with the param to start_internal and valid_adv_param.

@kapi-no give a try on this suggestion, having upto 4 max. parameters does not degrade performance, and the parameter serves as an auto-variable anyway.

Reply:
> 3. copy param to the local variable in the bt_conn_create_slave_le and set the implicit options

Isn't it the same as my initial suggestion?

Reply:
Not quite - I think the idea is to copy the options, not the full parameters struct

Reply:
..not that there's a big difference